backend/usecase/creation.go

```diff
@@ -78,3 +79,88 @@ func (u *CreationUsecase) TextCreation(ctx context.Context, req *domain.TextReq,
 	}
 	return nil
 }
+
+func (u *CreationUsecase) TabComplete(ctx context.Context, req *domain.CompleteReq) (string, error) {
+	// For FIM (Fill in Middle) style completion, we need to handle prefix and suffix
+	if req.Prefix != "" || req.Suffix != "" {
+		// System prompt for article continuation
+		systemPrompt := `
+你是集成在编辑器里的“内联续写”模型。任务是在光标处续写用户的文章，使其自然衔接并提升表达质量。
+
+严格遵循：
+1) 仅输出续写内容；不要复述已给文本；不要解释、不要前后标记、不要问答式措辞。
+2) 充分利用上下文：<PREFIX> 是光标前文本，<SUFFIX> 是光标后文本（如有）。
+   - 与 <PREFIX> 连贯一致，不改变已确立的观点、时间线、叙述视角与称谓。
+   - 不与 <SUFFIX> 冲突；如需要，使用最少字数平滑过渡到 <SUFFIX>。
+3) 维持原文语言与风格：语言保持与 <PREFIX> 一致（默认中文）；语气={tone}；受众={audience}；体裁={genre}。
```

Comment:
系统提示中包含了详细的指导原则，但这些原则可能需要根据具体的应用场景进行调整。建议添加机制允许动态调整这些原则，或者至少提供配置选项。

```suggestion
// 考虑将指导原则参数化，允许动态调整
systemPrompt := generateSystemPrompt(tone, audience, genre)